build.sbt

```diff
@@ -343,6 +343,28 @@ lazy val iceberg = (project in file("iceberg"))
     assembly / assemblyJarName := s"${name.value}_${scalaBinaryVersion.value}-${version.value}.jar",
     assembly / logLevel := Level.Info,
     assembly / test := {},
+    assembly / assemblyExcludedJars := {
+      val excludes = Seq("delta/spark", "delta/storage")
+      val cp = (assembly / fullClasspath).value
+
+      // Return `true` when we want the jar `f` to be excluded from the assembly jar
+      cp.filter { f =>
+        val result = excludes.exists(exclude => f.data.getPath.contains(exclude))
```

Comment:
is each value in `cp` a path to a class file or JAR file?

Reply:
Here's the complete input:

```
assemblyExcludedJars :: classes -> /Users/scott.sandre/delta/iceberg/target/scala-2.12/classes
assemblyExcludedJars :: classes -> /Users/scott.sandre/delta/spark/target/scala-2.12/classes
assemblyExcludedJars :: classes -> /Users/scott.sandre/delta/storage/target/classes
assemblyExcludedJars :: iceberg-shaded_2.12-3.0.0-SNAPSHOT.jar -> /Users/scott.sandre/delta/icebergShaded/target/scala-2.12/iceberg-shaded_2.12-3.0.0-SNAPSHOT.jar
assemblyExcludedJars :: scala-library-2.12.15.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.15/scala-library-2.12.15.jar
assemblyExcludedJars :: scala-collection-compat_2.12-2.1.1.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-collection-compat_2.12/2.1.1/scala-collection-compat_2.12-2.1.1.jar
assemblyExcludedJars :: caffeine-2.9.3.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/github/ben-manes/caffeine/caffeine/2.9.3/caffeine-2.9.3.jar
assemblyExcludedJars :: antlr4-runtime-4.9.3.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/antlr/antlr4-runtime/4.9.3/antlr4-runtime-4.9.3.jar
assemblyExcludedJars :: checker-qual-3.19.0.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/checkerframework/checker-qual/3.19.0/checker-qual-3.19.0.jar
assemblyExcludedJars :: error_prone_annotations-2.10.0.jar -> /Users/scott.sandre/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/google/errorprone/error_prone_annotations/2.10.0/error_prone_annotations-2.10.0.jar
```